src/expr/src/relation/func.rs

```diff
@@ -748,27 +820,113 @@ where
                     if offset > start_offset {
                         Datum::Null
                     } else {
-                        datums[end_idx].0
+                        args[end_idx]
                     }
                 } else {
-                    datums[end_idx].0
+                    args[end_idx]
                 }
             }
             // Forbidden during planning
             WindowFrameBound::UnboundedPreceding => unreachable!(),
         };
+        results.push(last_value);
+    }
+    results
+}
+
+/// Executes `FusedValueWindowFunc` on a reduction group.
+/// The expected input is in the format of `[((OriginalRow, (Args1, Args2, ...)), OrderByExprs...)]`
+/// where `Args1`, `Args2`, are the arguments of each of the fused functions. For functions that
+/// have only a single argument (first_value/last_value), these are simple values. For functions
+/// that have multiple arguments (lag/lead), these are also records.
```

Comment:
Yes, it's true that this function can handle multiple different types at each argument, and this couldn't be represented by our type system. However, the union fortunately doesn't need to be represented in the type system, because we know the actual type when looking at a specific query.

One way to think about this situation is that `AggregateFunc::FusedValueWindowFunc` is kind of a template function, in that it's parameterized by the functions that got fused. `fused_value_window_func` is an implementation of all possible instantiations of this template function.